components/brave_wallet/browser/network_manager.cc

```diff
@@ -29,80 +36,128 @@
+            {mojom::kOptimismMainnetChainId, "optimism-mainnet"},
+            {mojom::kBaseMainnetChainId, "base-mainnet"},
+            {mojom::kAvalancheMainnetChainId, "avalanche-mainnet"},
+            {mojom::kBnbSmartChainMainnetChainId, "bsc-mainnet"},
+
+            // SVM chains.
+            {mojom::kSolanaMainnet, "solana-mainnet"},
+
+            // Bitcoin chains.
+            {mojom::kBitcoinMainnet, "bitcoin-mainnet"},
+
+            // Cardano chains.
+            {mojom::kCardanoMainnet, "cardano-mainnet"},
+            {mojom::kCardanoTestnet, "cardano-preprod"},
+        },
+        CaseInsensitiveCompare());
 
 constexpr char kGanacheLocalhostURL[] = "http://localhost:7545/";
 constexpr char kSolanaLocalhostURL[] = "http://localhost:8899/";
 constexpr char kFilecoinLocalhostURL[] = "http://localhost:1234/rpc/v0";
 
-const std::string GetChainSubdomain(std::string_view chain_id) {
-  static base::NoDestructor<base::flat_map<std::string, std::string>>
-      subdomains({
-          // EVM chains.
-          {mojom::kMainnetChainId, "ethereum-mainnet"},
-          {mojom::kSepoliaChainId, "ethereum-sepolia"},
-          {mojom::kPolygonMainnetChainId, "polygon-mainnet"},
-          {mojom::kOptimismMainnetChainId, "optimism-mainnet"},
-          {mojom::kBaseMainnetChainId, "base-mainnet"},
-          {mojom::kAvalancheMainnetChainId, "avalanche-mainnet"},
-          {mojom::kBnbSmartChainMainnetChainId, "bsc-mainnet"},
-
-          // SVM chains.
-          {mojom::kSolanaMainnet, "solana-mainnet"},
-
-          // Bitcoin chains.
-          {mojom::kBitcoinMainnet, "bitcoin-mainnet"},
-
-          // Cardano chains.
-          {mojom::kCardanoMainnet, "cardano-mainnet"},
-          {mojom::kCardanoTestnet, "cardano-preprod"},
-      });
-
-  std::string chain_id_lower = base::ToLowerASCII(chain_id);
-  if (subdomains->contains(chain_id_lower)) {
-    return subdomains->at(chain_id_lower);
+enum class ToLowerCaseReason {
+  kGetURLForKnownChainId,
+  kGetCurrentChainIdFromPrefs,
+  kIsEip1559Chain,
+  kSetEip1559ForCustomChain,
+  kGetHiddenNetworks,
+  kAddHiddenNetwork
+};
+
+// A helper to check if there are any cases where the chain id is not lowercase,
+// to investigate why these conversions are required in the first place.
+// TODO(https://github.com/brave/brave-browser/issues/46940): Adding these
+// dumps in all places where this conversion is being done in this file to
+// better understand why this conversion is required in the first place, and
+// if we can completely eliminate them.
+std::string MakeChainIdLowerCase(std::string_view chain_id,
+                                 ToLowerCaseReason reason) {
+  if (!std::ranges::any_of(chain_id, base::IsAsciiUpper<char>)) {
+    return std::string(chain_id);
   }
 
-  return "";
+  // Only dumbing for M138 so it doesn't keep rolling if we forget about it
+  // (hopefully we won't though).
+  if (BASE_CHECK_VERSION_INTERNAL ==
```

Comment:
This is a bit naughty because it kinda violates the style:

> Use namespaces with "internal" in the name to document parts of an API that should not be mentioned by users of the API.

Although in this case it is not a namespace but rather just something tagged as internal. I can use `ersion_info::GetMajorVersionNumber()` if you prefer, but I had used this one because it seem much lighter to pull in.

Reply:
I've actually fixed it to use `version_info::GetMajorVersionNumberAsInt()`.